File: src/main/java/org/iuc/mobileteam/iucnewsappbackend/dtos/AnnouncementsDto.java
package org.iuc.mobileteam.iucnewsappbackend.dtos;

import org.bson.types.ObjectId;
import org.iuc.mobileteam.iucnewsappbackend.models.AnnouncementsEntity;

import java.util.Date;

public record AnnouncementsDto (
        String id,
        String title,
        String content,
        int readCount,
        Date createdAt,
        String originURL
) {
   public AnnouncementsDto(AnnouncementsEntity a){
       this(a.getId().toHexString(), a.getTitle(), a.getContent(), a.getReadCount(), a.getCreatedAt(), a.getOriginURL());
   }
   public AnnouncementsEntity toAnnouncementsEntity(){
       ObjectId _id = id == null ? new ObjectId() : new ObjectId(id);
         return new AnnouncementsEntity(_id, title, content, readCount, createdAt, originURL);
   }
}

File: src/main/kotlin/com/iucmobileteam/iucnewsbackend/dtos/AnnouncementsDto.kt
package com.iucmobileteam.iucnewsbackend.dtos

import com.iucmobileteam.iucnewsbackend.entity.AnnouncementsEntity
import java.util.Date

data class AnnouncementsDto (
    val id: Long? = null,
    val title: String? = null,
    val content: String? = null,
    val readCount: Int? = null,
    val publishDate: Date? = null,
    val originURL: String? = null
) {
    constructor(announcement: AnnouncementsEntity): this(
        id = announcement.id,
        title = announcement.title,
        content = announcement.content,
        readCount = announcement.readCount,
        publishDate = announcement.publishDate,
        originURL = announcement.originURL
    )

    fun toAnnouncementsEntity(): AnnouncementsEntity {
        return AnnouncementsEntity(
            id = this.id,
            title = this.title,
            content = this.content,
            readCount = this.readCount,
            publishDate = this.publishDate,
            originURL = this.originURL
        )
    }
}